src/app-host/app-host.js

```diff
@@ -5,93 +5,184 @@ var Messages = require('messages');
+    };
 
     // default Windows bootstrap function tries to load WinJS which is not
     // available and not required in simulation mode so we override bootstrap
     if (cordova.platformId === 'windows') {
-        cordova.require('cordova/platform').bootstrap = function () {
+        platformBootstrap = function () {
             cordova.require('cordova/modulemapper')
                 .clobbers('cordova/exec/proxy', 'cordova.commandProxy');
 
-            cordova.require('cordova/channel').onNativeReady.fire();
         };
     }
-}
 
-function getCordova() {
-    return cordova;
-}
+    channel = cordova.require('cordova/channel');
+
+    // define our own channel to delay the initialization until sim-host tells
+    // us everything's ready (fired in 'start' event handler).
+    channel.createSticky('onCordovaSimulateReady');
+    channel.waitForInitialization('onCordovaSimulateReady');
+
+    // firing of onNativeReady is delayed until SIM_HOST tells us it's ready
+    socket.once('init', function () {
+        // sim-host is ready, register exec handlers, fire onNativeReady and send
+        // the list of plugins
+        socket.on('exec-success', function (data) {
+            console.log('exec-success:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.success) {
+                execCacheInfo.success(data.result);
+            }
+        });
+
+        socket.on('exec-failure', function (data) {
+            console.log('exec-failure:');
+            console.log(data);
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo.fail) {
+                execCacheInfo.fail(data.error);
+            }
+        });
 
-socket.on('exec-success', function (data) {
-    console.log('exec-success:');
-    console.log(data);
-    var execCacheInfo = execCache[data.index];
-    if (execCacheInfo.success) {
-        execCacheInfo.success(data.result);
-    }
-});
+        socket.on('start-live-reload', function () {
+            livereload.start(socket);
+        });
 
-socket.on('exec-failure', function (data) {
-    console.log('exec-failure:');
-    console.log(data);
-    var execCacheInfo = execCache[data.index];
-    if (execCacheInfo.fail) {
-        execCacheInfo.fail(data.error);
-    }
-});
+        socket.on('init-telemetry', function (data) {
+            telemetry.init(socket);
+        });
 
-socket.on('start-live-reload', function () {
-    livereload.start(socket);
-});
+        socket.on('init-xhr-proxy', function (data) {
+            require('xhr-proxy').init(); 
+        });
 
-socket.on('init-telemetry', function (data) {
-    telemetry.init(socket);
-});
+        socket.on('init-touch-events', function (data) {
+            require('./touch-events').init();
+        });
 
-socket.on('init-xhr-proxy', function (data) {
-    require('xhr-proxy').init();
-});
+        channel.onNativeReady.fire();
```

Comment:
This is duplicated. I guess that you want to fire `onNativeReady` as you're doing below.